src/profile_data_exporter.cc

```diff
@@ -162,34 +162,40 @@ ProfileDataExporter::AddResponseTimestamps(
 
 void
 ProfileDataExporter::SetValueToJSON(
-    rapidjson::Value& json, const size_t index, const uint8_t* buf,
-    const size_t byte_size, const std::string& data_type)
+    rapidjson::Value& json, const size_t index, const std::vector<uint8_t>& buf,
+    const std::string& data_type)
 {
+  auto data_type_size_opt = GetDataTypeSize(data_type);
+  if (!data_type_size_opt || (index * *data_type_size_opt >= buf.size())) {
+    std::cerr << "Index out of bounds." << std::endl;
+    return;  // Handle error appropriately
```

Comment:
Did you mean TODO: in this comment?

Reply:
This comment I added for me to track when I was trying different things resulting in errors. I will remove it.